File: examples/03_function_curve.py
"""FunctionCurve: 函数曲线"""
import sys
import os
sys.path.insert(0, os.path.join(os.path.dirname(__file__), ".."))

from pysimdata.function import FunctionCurve

gen = FunctionCurve(func=__import__('numpy').sin, x_range=(0, 6.28), num_points=100, y_shape=50)
data = gen.generate()
print(f"FunctionCurve: {data.shape}")
gen.save_all("output/function_curve")

File: examples_cplus/03_function_curve.cpp
#include "pysimdata_cplus/pysimdata_cplus.h"

#include <iostream>

int main() {
    pysimdata_cplus::FunctionCurve gen(pysimdata_cplus::FunctionCurve::NameToFunc("sin"));
    gen.generate();
    std::string dir = gen.save("output/function_curve", "data", false);
    std::cout << "Saved to: " << dir << std::endl;
    return 0;
}
